Tweak std.algorithm.comparison.either docs and add to package doc table

JakobOvrum opened the pull request:
`either` is being introduced in 2.070, so this should probably be in the final release of that. What do you think @MartinNowak?

edit:

Should I file against `stable` in that case?

JakobOvrum commented on std/algorithm/comparison.d:
```diff
@@ -1962,24 +1962,24 @@ bool isPermutation(alias pred = "a == b", Range1, Range2)
-NOTE: Lazy parameters are currently, too restrictively, inferred by DMD to
-always throw eventhough they don't need to be. This makes it impossible to
-currently mark $(D either) as $(D nothrow). See issue at
-https://issues.dlang.org/show_bug.cgi?id=12647
```
DDoc was interpreting this as a section. `BUGZILLA` is nice in this case anyway.

quickfur replied:
Just write `_NOTE: ...` and it will suppress ddoc interpreting that as a section name.

JakobOvrum replied:
I was referring to the `https:` section.

quickfur commented on std/algorithm/comparison.d:
```diff
@@ -1962,24 +1962,24 @@ bool isPermutation(alias pred = "a == b", Range1, Range2)
 /**
-Get first parameter $(D p) that passes an $(D if (unaryFun!pred(p))) test.  If
-no parameter passes the test return the last.
+Get _first argument `a` that passes an `if (unaryFun!pred(a))` test.  If
```
Grammar nitpick: "Get **the** first argument ..."

quickfur commented on std/algorithm/comparison.d:
```diff
@@ -1962,24 +1962,24 @@ bool isPermutation(alias pred = "a == b", Range1, Range2)
-always throw eventhough they don't need to be. This makes it impossible to
-currently mark $(D either) as $(D nothrow). See issue at
-https://issues.dlang.org/show_bug.cgi?id=12647
+Note:
```
Now that I look at this again, I wonder if this section should be renamed to `Bugs:` instead of `Note:`, since it's arguably a bug that it can't be marked `nothrow`.

quickfur commented:
Other than that, I think we should merge this. Doc improvements are important.

quickfur commented:
ping
Let's finish this up and merge it.

quickfur commented:
ping @JakobOvrum

JakobOvrum commented:
@quickfur, the nitpicks you pointed out were already in master, and this patch was a substantial improvement as-is. Now this PR has stalled for literally a month over the absence of the definite article. I implore you not to do that again.

I've fixed the issues and rebased.

greenify commented on std/algorithm/comparison.d:
```diff
@@ -1962,24 +1962,24 @@ bool isPermutation(alias pred = "a == b", Range1, Range2)
 
 Returns:
-    The first parameter that passes the test $(D pred).
+    The _first argument that passes the test `pred`.
```
sorry for the question, but why is there an `_`?

DmitryOlshansky replied:
To avoid auto-detectin it as function parameter and highlighting based on that.

greenify commented:
> Now this PR has stalled for literally a month over the absence of the definite article. I implore you not to do that again.

Maybe we can figure out a way to prevent this in the future - see the discussion:
http://forum.dlang.org/thread/ljgzidjddfqpzjbnwemz@forum.dlang.org

DmitryOlshansky commented:
LGTM @quickfur ?

quickfur commented:
Let's just merge. This stalling is getting pretty ridiculous, and it's not as though we can't submit a followup PR to fix nitpicks, if any happen to slip through. Merge ahoy!

quickfur commented:
Auto-merge toggled on
